### PR TITLE
Structs and aliases generator in tgpu-cli

### LINKED ISSUE #362: Use `wgsl_reflect` to generate tgpu structures based on wgsl code as part of `@typegpu/tgpu-cli`

### PR DESCRIPTION
**Pull request by mhawryluk**
closes #362 

left to do:

- [x] arrays
- [x] atomics
- [x] tests

### CODE REVIEW

File: packages/tgpu-cli/gen.mjs [hunk 1]
```diff
@@ -1,25 +1,137 @@
...
+  const tgpuType = `${type instanceof StructInfo ? '' : 'd.'}${replaceWithAlias(type)}`;
+
+  return (
+    type.attributes?.reduce(
+      (acc, attribute) =>
+        ['align', 'size'].includes(attribute.name)
+          ? `d.${attribute.name}(${acc}, ${attribute.value})`
+          : acc,
+      tgpuType,
+    ) ?? tgpuType
```
Comment (iwoplaza) [inline packages/tgpu-cli/gen.mjs]: Love this use of `reduce` 💚

File: packages/tgpu-cli/gen.mjs [hunk 2]
```diff
@@ -1,25 +1,137 @@
...
+  return `${member.name}: ${generateType(member.type)},`;
+}
+
+/**
+ * @param { import('wgsl_reflect').TypeInfo } type
+ */
+function generateType(type) {
+  if (type instanceof ArrayInfo) {
+    return `d.arrayOf(${generateType(type.format)}, ${type.count > 0 ? type.count : LENGTH_VAR})`;
+  }
```
Comment (iwoplaza) [inline packages/tgpu-cli/gen.mjs]: Is exiting early here ignoring any attributes of array data types? (e.g. align or size)
Would be nice to have a test case of a custom aligned member, that is of an array type.

File: packages/tgpu-cli/package.json [hunk 3]
```diff
@@ -4,6 +4,10 @@
   "version": "0.0.0",
   "description": "A companion CLI to aid in developing with TypeGPU.",
   "license": "MIT",
+  "scripts": {
+    "test": "vitest run",
+    "dev:test": "vitest"
+  },
```
Comment (iwoplaza) [inline packages/tgpu-cli/package.json]: Running `test` and `dev:test` from the project root should work.

```suggestion
```

File: packages/tgpu-cli/tests/structs.test.ts [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+  velocity: vec2f,
+  isRed: u32,
+};`;
+
+    const expected = `
+const TriangleData = d.struct({
+  position: d.vec4f,
+  velocity: d.vec2f,
+  isRed: d.u32,
+});`.trim();
```
Comment (iwoplaza) [inline packages/tgpu-cli/tests/structs.test.ts]: Small suggestion, should have the same result.

```suggestion
    const expected = `\
const TriangleData = d.struct({
  position: d.vec4f,
  velocity: d.vec2f,
  isRed: d.u32,
});`;
```

File: packages/tgpu-cli/gen.mjs (hunk 2 above)
Comment (mhawryluk) [reply packages/tgpu-cli/gen.mjs]: good catch, fixed it

**Review by iwoplaza — APPROVED** [review]
Awesome job! ✨

**Pull request merged**